graspologic/cluster/rclust.py

```diff
@@ -0,0 +1,322 @@
+# Copyright (c) Microsoft Corporation and contributors.
+# Licensed under the MIT License.
+
+import numpy as np
+from sklearn.base import BaseEstimator
+from sklearn.utils import check_array
+from sklearn.utils.validation import check_is_fitted
+
+from anytree import NodeMixin
+
+from .autogmm import AutoGMMCluster
+from .kclust import KMeansCluster
+
+
+def _check_common_inputs(min_components, max_components, cluster_kws):
```

Comment:
@bdpedigo do you think adding "embed_kws" so that this function can be called by HSBMEstimator would be helpful?

Reply:
yes but let's do that in next PR, in general, i try to avoid engineering for the future that explicitly (we may change our minds later)